### PR TITLE
Chore writing docs section code snippets

### PR DESCRIPTION
**Pull request by jonniebigodes**
This pull request adds the snippets for the writing docs section:

What was done:

- Updated the markdown files to include the snippets based on the file naming and structure decided.
- Created the snippets (react,react ts, angular)

Feel free to provide feedback

### CODE REVIEW

**Review by tmeasday — CHANGES_REQUESTED** [review]
Great stuff @jonniebigodes -- some questions/notes on this:

1. A lot of these documentation examples (especially MDX) have a big slab of a code snippet with one line of react code in it. To me it feels unmaintainable to have a otherwise identical copy for each framework, but I think we need to call it out somehow. We could either:

  a. Have a comment above the react code: `// This is your Story template function, shown here in React`
  b. Comment out the code: `export const Template = /* your Story template function, as normal */;`
  c. Some complex solution where we inject code or something.

I'd be inclined to go with a. as I think folks might not know what b. means, and c. isn't going to happen.

2. What do you think about being a bit loose about TS snippets when the only difference is one single `as Meta` declaration? Sort of similar to the same question in the angular snippets, I'm not sure it is worth the maintenance burden in these cases where the snippets are so close to identical, but I'm happy to be guided by you.

3. It seems like in some cases you include the file name in the snippet and in others you don't? I think we want to always have the file name, right?

File: docs/snippets/common/storybook-preview-prepareforinline.js.mdx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+```js
+// .storybook/preview.js
+
+import React from 'react';
+import { render } from 'react-dom';
+import toReact from '@egoist/vue-to-react';
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/snippets/common/storybook-preview-prepareforinline.js.mdx:6]: @tmeasday the latter, this is more of a "example", like you put it, for demonstration purposes, should the reader be willing to add it to other framework he/she will have to adjust accordingly.

**Comment by jonniebigodes** [comment]
@tmeasday this was a section that still needs a bit of adjustments in order to convey the right message, hence why i left it like so and made the pull request so that we could see where we could improve it.

To answer your feedback.

Regarding this item:


Regarding this:
> A lot of these documentation examples (especially MDX) have a big slab of a code snippet with one line of react code in it. To me it feels unmaintainable to have a otherwise identical copy for each framework, but I think we need to call it out somehow. We could either:
a. Have a comment above the react code: // This is your Story template function, shown here in React
b. Comment out the code: export const Template = /* your Story template function, as normal */;
c. Some complex solution where we inject code or something.

We should go with option a, for sure. 

Also i left the mdx snippets for this section inside the common folder, despite being aware that they are based on the react implementation exactly for that purpose you brought up.

> What do you think about being a bit loose about TS snippets when the only difference is one single as Meta declaration? Sort of similar to the same question in the angular snippets, I'm not sure it is worth the maintenance burden in these cases where the snippets are so close to identical, but I'm happy to be guided by you.

The examples here were obtained from default content that is added when the user adds Storybook to his/hers project and with that i added them as to give the same experience that is provided and safe guard us from having issues arise in the mono repo mentioning that the docs don't match. But in the ts files if we could leave out and don't make too much of a impact fine by me. It will be better for us.

Regarding this item:

> It seems like in some cases you include the file name in the snippet and in others you don't? I think we want to always have the file name, right? 

I'll adjust accordingly shortly.

Feel free to provide feedback

**Comment by tmeasday** [comment]
> But in the ts files if we could leave out and don't make too much of a impact fine by me. It will be better for us.

Happy to be guided by you here. I don't know which is best, honestly. We can always change our mind later.

---

I'm gonna merge this PR so can you make any adjustments in another PR to `next` (will merge to `next` soon also).

**Pull request merged**